compactor/src/handler.rs

```diff
@@ -414,6 +406,167 @@ async fn compact_hot_partitions(compactor: Arc<Compactor>) -> usize {
     n_candidates
 }
 
+// For a given list of hot partition candidates and a memory budget, compute memory needed to compact each one
+// and compact as many of them in parallel as possible until all candidates are compacted
+async fn compact_hot_partition_candidates(
+    compactor: Arc<Compactor>,
+    mut candidates: Vec<PartitionCompactionCandidateWithInfo>,
+    table_columns: HashMap<TableId, Vec<ColumnTypeCount>>,
+) {
+    let mut remaining_budget_bytes = compactor.config.memory_budget_bytes;
+    let mut parallel_compacting_candidates = vec![];
```

Comment:
This should use `Vec::with_capacity`... I think this is the right capacity?

```suggestion
    let mut parallel_compacting_candidates = Vec::with_capacity(candidates.len());
```